### ISSUE TITLE
Handle fallout from mixin adjustments to biolink model

### ISSUE THREAD

**Issue by kvarforl** [RTXteam/RTX#1155]
I thought I would open a new issue for this since #1148 is getting long and this is more of a follow up topic.

**TLDR from #1148:** Some biolink node categories got changed to mixins and as such are no longer valid node categories. The fix for KG2.4.0 was to change all occurences of `occurrent` and `activity and behavior` in  curies_to_categories.yaml to be `named_thing`.

I'm not quite sure how to use mixins in KG2 (or if we even need to), and from discussing with Richard,  it sounds like the way mixins are incorporated into the biolink model is still new and liable to change. So for now, I'm thinking we avoid them until it stabilizes.

In the meantime, I think it makes sense to go through `curies_to_categories.yaml` and update any categories that are now  mixins to be the next best fitting category (including checking if the ones that already got changed to `named_thing` have a better fit).

I'll also update `validate_curies_to_categories_yaml.py` to fail if any of the categories are mixins.

**Comment by saramsey** [comment]
Thanks for the summary.

> Some biolink node categories got changed to mixins

Are there any others (that we were using in KG2) besides `occurrent` and `activity and behavior`?

**Comment by kvarforl** [comment]
I'm looking into it right now.  It seems like mixins also affect predicates. Just from inspection, there are 34 occurences of  "mixin: true" in biolink-model.yaml. 

The only other one that seems to effect KG2 is `regulates` (and therefore it's descendents, `positively regulates` and `negatively regulates`)  used to inherit from `affects`, but now doesn't inherit from anything.

**Comment by saramsey** [comment]
I think a key piece of information that we need from the Biolink/SRI team, is to know whether or not it is permissible for the `category` of a node to be an `occurrent`?  Or do they require that the category of a node is always a descendant of `named thing`?

**Comment by saramsey** [comment]
Richard B. confirms it: `activity and behavior` is not a legal biolink category for a KG node.

**Comment by saramsey** [comment]
So this actually simplifies our task, in some sense:

1. We use `named thing` in place of `occurrent` and `activity and behavior`
2. We lobby the SRI/Biolink team to see if the Biolink model could be evolved/tweaked so that we have something better than `named thing` to represent activity or occurent type nodes.

**Comment by saramsey** [comment]
For item (2) in the last post, I have logged this issue in the Biolink Github repo:

https://github.com/biolink/biolink-model/issues/578

**Comment by saramsey** [comment]
In `curies-to-categories.yaml`, I think `T053` (`behavior`), `T054` (`social behavior`), and `T055` (`individual behavior`) can be mapped to `phenomenon`, which would be a bit better than `named thing`.

**Comment by kvarforl** [comment]
@saramsey  should I close this issue, or leave it open as a reminder to follow up on the biolink issue?

**Comment by saramsey** [comment]
> @saramsey should I close this issue, or leave it open as a reminder to follow up on the biolink issue?

I'm not sure. What issue are you referring to?  Do you mean the potential issue that Biolink could break out build at any time, by changing some concrete class to a mixin abstract class?  Or some other issue.

**Comment by kvarforl** [comment]
Ah sorry for the lack of clarification! I was thinking that if anything happens with [biolink/biolink-model#578](https://github.com/biolink/biolink-model/issues/578), we'll need to remember to go back through and replace anything that best fits within `biolink:activity` to map to that.

**Comment by saramsey** [comment]
Hi @kvarforl, OK, the Biolink team has apparently added `biolink:activity` as a Biolink category. As soon as you see it showing up in the `biolink-model.yaml` file, can we start using it to annotate UMLS TUIs in `curies-to-categories.yaml` like `T052`, `T057`, `T058`, and `T062`?

**Comment by kvarforl** [comment]
I went through and updated anything that I had marked in this issue and #1148 as being formerly `activity and behavior` to be `activity`.

```
lindsey@lindsey-ThinkPad-T560:~/Research/kg2-code$ grep ": activity" curies-to-categories.yaml 
    MESH:D001520: activity
    NBO:0000313: activity
    NCIT:C16203: activity
    NCIT:C16326: activity
    NCIT:C43431: activity
    SNOMED:48176007: activity
    UMLS_STY:T051: activity 
    UMLS_STY:T052: activity
    UMLS_STY:T053: activity
    UMLS_STY:T054: activity
    UMLS_STY:T055: activity
    UMLS_STY:T056: activity
    UMLS_STY:T057: activity 
    UMLS_STY:T064: activity 
    UMLS_STY:T066: activity
```

**Comment by kvarforl** [comment]
Kg2.5.0 has this fix included. Closing the issue
